Upgraded latest version of flutter

mevluttosun opened the pull request:
Upgraded android sdk and gradle
Upgraded ios versions

Commit:
first commit

Commit:
change home page

Commit:
downgrade version dart

Commit:
fix background ios

haduyphuong committed:
update README.md

haduyphuong committed:
update README.md

mevluttosun committed:
refactored with newest flutter plugin settings. Upgraded sdk versions

github-actions[bot] committed:
add Podfile and Podfile.lock for CocoaPods integration; update xcconfig files to include Pods support

haduyphuong commented on ios/Resources/PrivacyInfo.xcprivacy:
Why you use there permissions?

mevluttosun replied:
It is default. I open default empty plugin then replace it with yours. If you want I can remove.

haduyphuong commented on example/pubspec.yaml:
```diff
@@ -1,13 +1,11 @@
-version: 1.0.0+1
 publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 
 environment:
```
I want the low version to be usable.

haduyphuong commented on example/pubspec.yaml:
```diff
@@ -1,13 +1,11 @@
 
 environment:
-  sdk: '>=2.16.0 <3.0.0'
+  sdk: ^3.5.4
```
I want the low version to be usable.

haduyphuong commented on example/.gitignore:
```diff
@@ -1,7 +1,6 @@
 # Miscellaneous
 *.class
 *.log
-*.lock
```
I think need keep rule .lock in this file

haduyphuong commented on LICENSE:
```diff
@@ -1,20 +1 @@
-The MIT License (MIT)
```
This file should not be deleted or left blank.

haduyphuong commented on example/android/app/src/main/AndroidManifest.xml:
```diff
@@ -31,4 +31,15 @@
             android:name="flutterEmbedding"
             android:value="2" />
     </application>
+    <!-- Required to query activities that can process text, see:
```
Why you can use permission PROCESS_TEXT

haduyphuong commented on example/pubspec.lock:
```diff
@@ -0,0 +1,283 @@
+# Generated by pub
```
I think need remove this file

haduyphuong commented on ios/flutter_lifecycle_detector.podspec:
```diff
@@ -5,19 +5,25 @@
   s.name             = 'flutter_lifecycle_detector'
   s.version          = '0.0.1'
-  s.summary          = 'A plugin flutter detect app entering background or foreground mode.'
+  s.summary          = 'A new Flutter plugin project.'
```
You was remove description of package????

haduyphuong commented on pubspec.yaml:
```diff
@@ -1,20 +1,21 @@
-description: A plugin flutter detect app entering background or foreground mode.
-version: 0.0.6
-homepage: https://github.com/haduyphuong/flutter_lifecycle_detector
+description: "A new Flutter plugin project."
```
You was remove description of package???? You can not remove link github of this project.

haduyphuong commented on pubspec.yaml:
```diff
@@ -1,20 +1,21 @@
 name: flutter_lifecycle_detector
-description: A plugin flutter detect app entering background or foreground mode.
-version: 0.0.6
-homepage: https://github.com/haduyphuong/flutter_lifecycle_detector
```
You was remove description of package????

haduyphuong commented on pubspec.yaml:
```diff
@@ -1,20 +1,21 @@
 environment:
-  sdk: '>=2.16.0 <3.0.0'
-  flutter: ">=2.5.0"
+  sdk: ^3.5.4
```
I want the low version to be usable.

mevluttosun committed:
Added rule to exclude *.lock files in both the main and example directories

mevluttosun committed:
Re-added the MIT License to the project

mevluttosun committed:
Decreased the minimum SDK version to 2.17.0 and Flutter version to 2.5.0 in pubspec.yaml.
Adjusted the example's SDK version to 2.17.0 to align with the main project settings.

mevluttosun committed:
Removed PROCESS_TEXT permission from AndroidManifest.xml in the example app

mevluttosun committed:
Reverted package description and removed unnecessary comments in flutter_lifecycle_detector.podspec

github-actions[bot] committed:
Removed unnecessary privacy information from PrivacyInfo.xcprivacy

haduyphuong commented:
I was update new sdk Flutter
